File: shared/contracts/python/v1/process.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

JobStatus = Literal["queued", "running", "completed", "failed"]


class ProcessNoteRequest(BaseModel):
    note_id: str = Field(min_length=1)
    content_text: str = Field(min_length=1)
    content_hash: str = Field(min_length=1)
    updated_at: str = Field(min_length=1)


class ProcessNoteResponse(BaseModel):
    job_id: str
    status: Literal["queued"]


class ProcessStatusResponse(BaseModel):
    job_id: str
    status: JobStatus
    created_at: str
    updated_at: str
    error: str | None = None

File: shared/contracts/ts/v1/process.ts
export type JobStatus = "queued" | "running" | "completed" | "failed";

export interface ProcessNoteRequest {
  note_id: string;
  content_text: string;
  content_hash: string;
  updated_at: string;
}

export interface ProcessNoteResponse {
  job_id: string;
  status: "queued";
}

export interface ProcessStatusResponse {
  job_id: string;
  status: JobStatus;
  created_at: string;
  updated_at: string;
  error?: string | null;
}
